### ISSUE TITLE
[no-var-requires] False negative if require result is passed to a function

### ISSUE THREAD

**Issue by theneva** [typescript-eslint/typescript-eslint#665]
**Repro**

The provided code can also be found in the minimal reproduction repo at https://github.com/theneva/no-var-requires-trick.

I use the @typescript-eslint/recommended plugin with no configuration (and disable @typescript-eslint/no-unused-vars to reduce noise for the repro):

```JSON
{
  "extends": ["plugin:@typescript-eslint/recommended"],
  "rules": {
    "@typescript-eslint/no-unused-vars": "off"
  }
}
```

I can then trick the rule by passing the result of the `require('fs')` call into a function:

```TS
// your repro code case
function trick(a: unknown) {
  return a;
}

const fs = trick(require('fs'));
```

**Expected Result**

The rule should report this as an error, as I am using the returned value of a `require(…)` call.

**Actual Result**

The rule does not consider this to be an error.

**Additional Info**

When my TS code looks like this (with the same eslint config), the rule reports one assignment and ignores the other:

```ts
function trick(a: unknown) {
  return a;
}

const fs = trick(require('fs'));

const fs2 = require('fs');
```

The output looks as follows (reporting `fs2` but not `fs`):

```
$ yarn run eslint .

/private/tmp/no-var-requires-trick/index.js
  5:11  error  Require statement not part of import statement  @typescript-eslint/no-var-requires

✖ 1 problem (1 error, 0 warnings)
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.11.0` |
| `@typescript-eslint/parser`        | `1.11.0` |
| `TypeScript`                       | `3.5.2` |
| `ESLint`                           | `6.0.1` |
| `node`                             | `11.10.1` |
| `npm`                              | `N/A` (`yarn` version `1.17.0`) |

**Comment by bradzacher** [comment]
I mean technically, this is working as intended, as you are not using var var require.

Happy to accept a PR for this.

**Comment by JayaKrishnaNamburu** [comment]
Hi @bradzacher I would like to try this and raise PR. But I tried cloning the repo and repro using ``yarn run eslint`` and but it does not throw any error for both the cases for me

**Comment by JayaKrishnaNamburu** [comment]
Hi @bradzacher I give a PR, it fixes the current. It fixes the current issue, would like to know if there are any edge cases are any other preferred way to handle this.
